amethyst_network/src/network_socket.rs

```diff
@@ -0,0 +1,193 @@
+//! The network send and receive System
+
+use super::{deserialize_event, send_event, ConnectionState, NetConnection, NetEvent, NetFilter};
+use amethyst_core::specs::{Join, Resources, System, SystemData, WriteStorage};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::clone::Clone;
+use std::io::{Error, ErrorKind};
+use std::net::IpAddr;
+use std::net::SocketAddr;
+use std::net::UdpSocket;
+use std::str;
+use std::str::FromStr;
+
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::thread;
+
+enum InternalSocketEvent<E> {
+    SendEvents {
+        target: SocketAddr,
+        events: Vec<NetEvent<E>>,
+    },
+    Stop,
+}
+
+struct RawEvent {
+    pub byte_count: usize,
+    pub data: Vec<u8>,
+    pub source: SocketAddr,
+}
+
+// If a client sends both a connect event and other events,
+// only the connect event will be considered valid and all others will be lost.
+
+/// The System managing the network state and connections.
+/// The T generic parameter corresponds to the network event type.
+/// Receives events and filters them.
+/// Received events will be inserted into the NetReceiveBuffer resource.
+/// To send an event, add it to the NetSendBuffer resource.
+///
+/// If both a connection (Connect or Connected) event is received at the same time as another event from the same connection,
+/// only the connection event will be considered and rest will be filtered out.
+// TODO: add Unchecked Event type list. Those events will be let pass the client connected filter (Example: NetEvent::Connect).
+// Current behaviour: hardcoded passthrough of Connect and Connected events.
```

Comment:
The comments and docs content are fine, but the comment is split in half by the doc in the middle xD

Reply:
line 33